src/XPurchase/VirtoCommerce.XPurchase/Commands/AddCartItemCommandHandler.cs

```diff
@@ -27,6 +27,12 @@ await cartAggregate.AddItemAsync(new NewCartItem(request.ProductId, request.Quan
                 CartProduct = product
             });
 
+            // Prevent saving invalid cart
+            if (!cartAggregate.IsValid)
```

Comment:
we can't prevent to save the cart contains validation errors, because many of them are recoverable and may require the user's attention, for example "Product no longer available", or a "price changed", all of these errors are required confirmation from the user, and we can't restrict to save cart with validation errors if the user decided to save some action for one of the products that has the validation error.